tests/functional/testSettings.js

```diff
@@ -808,4 +809,49 @@ describe('Settings', function() {
     await vpn.clickOnQuery(queries.screenSettings.SIGN_OUT.visible());
     await vpn.waitForInitialView();
   });
+
+  it('Checking Developer Menu Reset and Quit', async () => {
+    // magically unlock dev menu
+    await vpn.setSetting('developerUnlock', 'true');
+
+    // navigate to Developer Menu
+    await getToGetHelpView();
+    await vpn.waitForQueryAndClick(
+        queries.screenGetHelp.DEVELOPER_MENU.visible());
+
+    // click "reset and quit" 6 times, test will fail if app quits early
+    await vpn.waitForQuery(
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON.visible());
+    await vpn.scrollToQuery(
+        queries.screenDeveloperMenu.SCREEN,
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON);
+    await vpn.waitForQueryAndClick(
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON.visible());
+    await vpn.waitForQueryAndClick(
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON.visible());
+    await vpn.waitForQueryAndClick(
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON.visible());
+    await vpn.waitForQueryAndClick(
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON.visible());
+    await vpn.waitForQueryAndClick(
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON.visible());
+    // can't use waitForQueryAndClick for final click because it returns an
+    // error - as expected, we crashed the app - but that causes test to fail
+    await vpn.clickOnQueryAndAcceptAnyResults(
+        queries.screenDeveloperMenu.RESET_AND_QUIT_BUTTON.visible());
+
+    // Confirm the app quit
+    assert.equal(setup.vpnIsInactive(), true);
+
+    // relaunch app
+    await setup.startAndConnect();
```

Comment:
wasm seems to struggle a bit with restarting the client, so I've been excluding these kinds of tests via

``` JS
    if (this.ctx.wasm) {
      return;
    }
```

Reply:
Thanks for saving me some research on this one. Just added this in.